app/models/finder_schema.rb

```diff
@@ -2,6 +2,12 @@ class FinderSchema
   include ActiveModel::Model
   include ActiveModel::Attributes
   include ActiveModel::Serializers::JSON
+  extend ActiveModel::Callbacks
+
+  define_model_callbacks :update
+
+  before_update :reset_email_alerts
```

Comment:
Wouldn't this also reset the email alerts when we _don't_ want to reset them?
I.e. from the "edit metadata" or "edit filters/facets" forms?

Reply:
Yes - those other forms don't exist yet do they?

If/when they do we'll have to find another approach. Two spring to mind:

1. exclude these attributes when copying the attributes from the old schema into the proposed one. I didn't do this at first because I didn't want to put more logic into the controller. I don't feel strongly about that though
2. create some kind of update context option along the lines of Rails validation contexts to control which callbacks are applied

Any thoughts? There are probably other options as well

Reply:
The first form already exists - see https://specialist-publisher.publishing.service.gov.uk/admin/metadata/aaib-reports

Of the two proposals, the second sounds best IMO.

Reply:
That's the same form I'm working on though (you'll notice the Email Alerts section with the legend in the wrong place at the bottom 🤦 )

Appreciate we'll have to do the work anyway when we come to filters and facets - happy to do it now if you'd prefer

Reply:
Oh! I could have sworn we'd decided to make the email-signup-editing its own form. I guess not (or the designs have since been updated!).

Less of a reason to do it now if it's not going to break any existing forms. Just so long as we remember to bear it in mind for the facets form 👍 or you could do it now to reduce the risk of forgetting it.

Reply:
I suggest we make it a later if that's ok - no guarantees that we're going to get to filters and facets this quarter and I think it's going to require quite a lot of change to the finder schema model anyway, so this may need a rethink